llama-cpp-sys-2/build.rs

```diff
@@ -1,187 +1,41 @@
-            } else if env::var("HOST") == env::var("TARGET") {
-                build.flag("-mcpu=native");
-            }
-            build.flag("-pthread");
-        }
-    }
-
-    // https://github.com/ggerganov/llama.cpp/blob/191221178f51b6e81122c5bda0fd79620e547d07/Makefile#L133-L141
     if cfg!(target_os = "macos") {
-        assert!(!cublas_enabled, "CUBLAS is not supported on macOS");
-
-        println!("cargo:rustc-link-lib=framework=Metal");
-        println!("cargo:rustc-link-lib=framework=Foundation");
-        println!("cargo:rustc-link-lib=framework=MetalPerformanceShaders");
-        println!("cargo:rustc-link-lib=framework=MetalKit");
-
-        llama_cpp.define("_DARWIN_C_SOURCE", None);
-
-        // https://github.com/ggerganov/llama.cpp/blob/3c0d25c4756742ebf15ad44700fabc0700c638bd/Makefile#L340-L343
-        llama_cpp.define("GGML_USE_METAL", None);
-        llama_cpp.define("GGML_USE_ACCELERATE", None);
-        llama_cpp.define("ACCELERATE_NEW_LAPACK", None);
-        llama_cpp.define("ACCELERATE_LAPACK_ILP64", None);
-        println!("cargo:rustc-link-lib=framework=Accelerate");
-
-        metal_hack(&mut ggml);
-        ggml.include("./llama.cpp/ggml-metal.h");
+        metal_hack();
     }
 
-    if cfg!(target_os = "dragonfly") {
-        llama_cpp.define("__BSD_VISIBLE", None);
-    }
-
-    if cfg!(target_os = "linux") {
-        ggml.define("_GNU_SOURCE", None);
-    }
-
-    ggml.std("c11")
-        .include("./llama.cpp")
-        .file("llama.cpp/ggml.c")
-        .file("llama.cpp/ggml-alloc.c")
-        .file("llama.cpp/ggml-backend.c")
-        .file("llama.cpp/ggml-quants.c")
-        .define("GGML_USE_K_QUANTS", None);
-
-    llama_cpp
-        .define("_XOPEN_SOURCE", Some("600"))
-        .include("llama.cpp")
-        .std("c++11")
-        .file("llama.cpp/llama.cpp")
-        .file("llama.cpp/unicode.cpp");
-
-    // Remove debug log output from `llama.cpp`
-    let is_release = env::var("PROFILE").unwrap() == "release";
-    if is_release {
-        ggml.define("NDEBUG", None);
-        llama_cpp.define("NDEBUG", None);
-        if let Some(cuda) = ggml_cuda.as_mut() {
-            cuda.define("NDEBUG", None);
-        }
-    }
-
-    if let Some(ggml_cuda) = ggml_cuda {
-        println!("compiling ggml-cuda");
-        ggml_cuda.compile("ggml-cuda");
-        println!("compiled ggml-cuda");
-    }
-
-    println!("compiling ggml");
-    ggml.compile("ggml");
-    println!("compiled ggml");
-
-    println!("compiling llama");
-    llama_cpp.compile("llama");
-    println!("compiled llama");
+    let build = Config::new("llama.cpp")
+        .define(
+            "LLAMA_CUBLAS",
+            if cfg!(feature = "cublas") {
+                "ON"
+            } else {
+                "OFF"
+            },
+        )
+        .define("LLAMA_VULKAN", if cfg!(feature = "vulkan") {
+            "ON"
+        } else {
+            "OFF"
+        })
+        .define("BUILD_SHARED_LIBS", "ON")
```

Comment:
It would be appreciated to have a `static` feature for those who prefer static linking.